platform/journal.go

```diff
@@ -83,16 +92,18 @@ func (j *Journal) Start(ctx context.Context, m Machine) error {
 }
 
 func (j *Journal) Destroy() error {
-	var err error
+	var err multierror.Error
 	if j.cancel != nil {
 		j.cancel()
-		err = j.recorder.Wait()
+		if e := j.recorder.Wait(); e != nil {
+			err = append(err, e)
+		}
 	}
-	if err2 := j.journal.Close(); err == nil && err2 != nil {
-		err = err2
+	if e := j.journal.Close(); e != nil {
+		err = append(err, e)
 	}
-	if err2 := j.journalRaw.Close(); err == nil && err2 != nil {
-		err = err2
+	if e := j.journalRaw.Close(); e != nil {
```

Comment:
[Leaks the underlying File](https://golang.org/pkg/compress/gzip/#Writer.Close).